external-crates/move/crates/move-cli/src/base/new.rs

```diff
@@ -43,17 +37,43 @@ impl New {
         custom: &str, // anything else that needs to end up being in Move.toml (or empty string)
     ) -> anyhow::Result<()> {
         // TODO warn on build config flags
-        let Self { name } = self;
 
-        if !Identifier::is_valid(&name) {
-            return Err(anyhow!(
+        if !Identifier::is_valid(&self.name) {
+            bail!(
                 "Invalid package name. Package name must start with a lowercase letter \
                  and consist only of lowercase letters, numbers, and underscores."
-            ));
+            );
         }
 
-        let path = path.unwrap_or_else(|| Path::new(&name));
+        let path = path.unwrap_or_else(|| Path::new(&self.name));
         create_dir_all(path.join(SourcePackageLayout::Sources.path()))?;
+
+        self.write_move_toml(path, deps, addrs, custom)?;
+        self.write_gitignore(path)?;
+        Ok(())
+    }
+
+    /// add `build/*` to `{path}/.gitignore`
+    fn write_gitignore(&self, path: &Path) -> anyhow::Result<()> {
```

Comment:
Can we also add a test for this `write_gitignore` function to ensure that the file is being added?

Reply:
that's why this is a draft =)